### ISSUE TITLE
defaults.lua nil error

### ISSUE THREAD

**Issue by trappedinspacetime** [mpv-player/mpv#8795]
Ubuntu 20.04
mpv 0.33.0-128-gd3b7732aaa
[log file](https://0x0.st/-abn.txt)

I used to use [mpv-youtube-search](https://github.com/jgreco/mpv-youtube-search) script on Ubuntu 16.04 successfully last year.
I installed Ubuntu 20.04 recently. I built mpv and added that script again  but I get a crash this time. The script hasn't had any update since then.

          [  11.467][w][youtube_search] stack traceback:
          [  11.467][w][youtube_search] 	[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:872: in function 'parse_characters'
          [  11.467][w][youtube_search] 	[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:893: in function 'create_character_map'
          [  11.467][w][youtube_search] 	[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:910: in function 'calculate_width'
          [  11.467][w][youtube_search] 	[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:1386: in function 'get_ass'
          [  11.467][w][youtube_search] 	[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:1679: in function 'handler'
          [  11.467][w][youtube_search] 	mp.defaults:500: in function 'call_event_handlers'
          [  11.468][w][youtube_search] 	mp.defaults:534: in function 'dispatch_events'
          [  11.468][w][youtube_search] 	mp.defaults:493: in function <mp.defaults:492>
          [  11.468][w][youtube_search] 	[C]: in ?
          [  11.468][w][youtube_search] 	[C]: in ?
          [  11.468][f][youtube_search] Lua error: [string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:872: attempt to perform arithmetic on field '?' (a nil value)

I thought it might be related to changes in https://github.com/mpv-player/mpv/blob/master/player/lua/defaults.lua
I need help.

**Comment by avih** [comment]
> defaults.lua nil error

That's meaningless, because all scripts are called from `defaults.lua`, and `defaults.lua` will always appear at any error stack traces which a script has.. The actual issue is at `[string "/home/kenn/.config/mpv/scripts/youtube-sear..."]:872: in function 'parse_characters'`

> I used to use [mpv-youtube-search](https://github.com/jgreco/mpv-youtube-search) script

Other than its very bold message "!!! VERY BETA !!!", this script has not been updated for two years, while mpv has changed since.

We can't guarantee that scripts will remain compatible forever (even if we do try to avoid or at least minimize incompatible changes).

The bottom line is that script authors should keep their scripts up to date with mpv updates, and you should try to contact the script author and/or fix it yourself.

If this turns out to be an mpv issue, then you can open a new issue about that with a minimal way to reproduce it without 3rd party scripts.

**Comment by avih** [comment]
For posterity, adding reference to the script issue: https://github.com/jgreco/mpv-youtube-search/issues/1

**Comment by trappedinspacetime** [comment]
if I remove ['--vf-add=format=gray](https://github.com/jgreco/mpv-youtube-search/blob/master/youtube-search.lua#L804)', it doesn't crash and no error but when I enter a video name into search box, it doesn't present search results in thumbnails, instead it plays the first video it finds.

**Comment by avih** [comment]
> if I remove '--vf-add=format=gray', it doesn't crash and no error but when I enter a video name into search box

This is completely out of mpv control. It's a script issue, plain and simple.

We can't do anything about that unless you or someone else finds the underlying reason - and we conclude that it's indeed an mpv issue, but for now I have no reason to suspect that it's an mpv issue.
